src/Hangfire.Tags.SqlServer/SqlTagsServiceStorage.cs

```diff
@@ -166,13 +166,12 @@ private int GetJobCount(DbConnection connection, string[] tags, string stateName
 )
 select count(*)
 from [{_options.SchemaName}].Job j with (nolock)
-inner join cte on cte.Id = j.Id 
-left join [{_options.SchemaName}].State s with (nolock) on j.StateId = s.Id";
```

Comment:
The query doesn't use any data from State table. 
We should remove it to improve performance